### ISSUE TITLE
Homepage modal: Using a screen reader when closing modal announces header before skip link

### ISSUE THREAD

**Issue by laflannery** [department-of-veterans-affairs/va.gov-cms#12008]
## Describe the defect
Using a screen reader, when the homepage modal is closed, regardless of how it's closed (X, button, ESC key), before the skip link is announced, the first header in the footer (Veterans programs and services) is announced.

## To Reproduce
1. Turn on your screen reader  (i.e. NVDA)
2. Go to (when this is deployed I can add the staging homepage)
3. Close the modal by using the ESC key, the X in the corner or the "Not today..." button
4. Note that the screen reader announces "Veterans programs and services" then announces "Skip to content"
5. However, visually the Skip link is shown as focused immediately

## AC / Expected behavior

- [ ] When the modal is closed (using the X, the button or the ESC), the SR should announce the same element as what is visually focused - in this case the Skip to content link
- [ ] When resolved / verified in prod, also close https://github.com/department-of-veterans-affairs/va.gov-team/issues/51155

## Additional Context
@jtmst mentioned he thought he knew what was going on here and what would need to be looked at to resolve it:

> there was an issue with set-focus not working properly when closing the modal, a solution ive used for similar bugs is adding a 50ms delay before changing focus. I didnt anticipate it affecting screen readers this way.

### CMS Team
Please check the team(s) that will do this work.

- [ ] `Program`
- [ ] `Platform CMS Team`
- [ ] `Sitewide Crew`
- [ ] `⭐️ Sitewide CMS`
- [x] `⭐️ Public Websites`
- [ ] `⭐️ Facilities`
- [ ] `⭐️ User support`

**Comment by laflannery** [comment]
Options from discussion with Brian DeConinck:
- load the new div on modal open rather than modal close
- Move the modal from the footer to above new div
- Add a blank space above the new div that is not hidden in any way to there is something available for NVDA to recognize before going to the hidden elements

Still waiting to hear from Martha on if this will still be launching blocking even though it's a 3rd party bug

**Comment by jilladams** [comment]
@jtmst I think we ended up with dupe issues. Does the PR that merged today to move the modal above skip link for #12045 also help resolve this issue?

**Comment by jtmst** [comment]
> @jtmst I think we ended up with dupe issues. Does the PR that merged today to move the modal above skip link for #12045 also help resolve this issue?

Yes thats correct.

**Comment by jilladams** [comment]
Great - moving to PO Review, and we can close both once the fix is verified.

**Comment by laflannery** [comment]
@jilladams I reviewed this yesterday and spoke to Josh. The feedback we discussed was merging [the other PR](https://github.com/department-of-veterans-affairs/vets-website/pull/23010) into stage so that everything can be tested together, because currently the skip link is visible again which I know we don't want. Unless there is a separate issue for the skip link tracking that PR?

**Comment by jilladams** [comment]
Noted, and also noting: these changes are all now present on Staging for verification, where Skip link is no longer visually present when closing the modal. 

@laflannery for verifying with NVDA, note that we merged some logic that will only show the modal to 25% of users. So to force the modal to appear if it doesn't on its own, you will need to edit a local storage variable:  `show-homepage-soft-launch-modal` and set that to a number between 1 - 25.

**Comment by laflannery** [comment]
@jilladams I just tested everything on stage and I have summarized everything below:
- When closing the modal, the footer is no longer being announced, which was the main issue and source of confusion.
- Using NVDA, there is still some inconsistency in whether or not the Skip link is announced first or the the "Current Homepage" div is announced first (this seems like it might be dependent on how quickly the modal is closed?)
- However, when using JAWS, "Current Homepage" was always announced first.
- The official 508 office policy is that JAWS is used as part of their auditing activities for all applications

With all of that in mind, I would say that this is is an acceptable solution for the NVDA bug.
